### PR TITLE
Downgrade google-api-core to mitigate #30927

### PR DESCRIPTION
**Pull request by tvalentyn**
A change in google-api-core 2.17.0 triggered a regression introduced in grpcio==1.58.0, which will be fixed in grpcio==1.63.0 and possibly in grpcio==1.62.2.  The regression affects Beam 2.55.0 (#30927 #30867).

This temporary change will help avoid the dependency version combination that might trigger the failure until the grpc fix is available.

### CODE REVIEW

**Comment by tvalentyn** [comment]
R: @damccorm

File: sdks/python/setup.py [hunk 1]
```diff
@@ -436,7 +436,9 @@ def get_portability_package_data():
           ],
           'gcp': [
               'cachetools>=3.1.0,<6',
-              'google-api-core>=2.0.0,<3',
+              # Temporary workaround until grpcio releases a fix for
+              # https://github.com/grpc/grpc/issues/36265
+              'google-api-core>=2.0.0,!=2.17.*,!=2.18.*,<3',
```
Comment (damccorm) [inline sdks/python/setup.py:441]: Is this going to be fixed in google-api-core, or should we just pin to <2.17 until the fix is merged?

Comment (tvalentyn) [reply sdks/python/setup.py:441]: it will be fixed in grpcio.

Comment (tvalentyn) [reply sdks/python/setup.py:441]: a lower bound restriction is prone to cause dependency resolution down the road.

Comment (tvalentyn) [reply sdks/python/setup.py:441]: a better fix would be to exclude the broken versions of grpcio, but that would revert a recent change to enable Py 3.12 support and I'd prefer to let people to experiment with Py 3.12 as early as possible to gather more feedback even before we officially support it.

Comment (tvalentyn) [reply sdks/python/setup.py:441]: if grpc team produces a fix in 1.62.2, we can revert this change,  regenerate the dependencies and cherry-pick that for 2.56.0.

Comment (damccorm) [reply sdks/python/setup.py:441]: My concern is mostly that if google-api-core releases 2.19 and people are on old grpcio this problem will still exist. I think you're probably right, that is the best option still though.

Hopefully the fix comes in soon and we can cherry-pick in a better patch

**Review by damccorm — APPROVED** [review]

**Pull request merged**